ajsonrpc/manager.py

```diff
@@ -37,9 +37,13 @@ async def get_response_for_request(self, request: JSONRPC20Request) -> Optional[
             )
         else:
             try:
-                result = await method(*request.args, **request.kwargs) \
-                    if inspect.iscoroutinefunction(method) \
-                    else method(*request.args, **request.kwargs)
+                # Compatible with coroutine cyfunction or other kind of functions which return coroutines
+                # but are not coroutine functions
+                co_or_ret = method(*request.args, **request.kwargs)
+                # Maybe `inspect.isawaitable()` is better?
```

Comment:
could you please elaborate on this comment? What are pros/cons of `inspect.isawaitable`? If we pick it, do we need this comment?